MapboxDirections/MBIntersection.swift

```diff
@@ -0,0 +1,108 @@
+import Foundation
+
+@objc(MBIntersection)
+public class Intersection: NSObject, NSSecureCoding {
+    
+    /**
+     Index into bearings/entry array.
+     
+     Used to calculate the bearing just before the turn. Namely, the clockwise angle from true north to the direction of travel immediately before the maneuver/passing the intersection. Bearings are given relative to the intersection. To get the bearing in the direction of driving, the bearing has to be rotated by a value of 180. The value is not supplied for depart maneuvers.
+    */
+    public let approachIndex: Int?
```

Comment:
Unfortunately, `Int?` won’t bridge to Objective-C, so this will have to be `Int` instead. Set it to `-1` by default (for the v4 case). Unfortunate, but this is one of the tradeoffs of supporting Objective-C.